components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenManagementDAOImpl.java

```diff
@@ -97,39 +98,37 @@ public RefreshTokenValidationDataDO validateRefreshToken(String consumerKey, Str
 
         try {
             String driverName = connection.getMetaData().getDriverName();
-            if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
-                if (driverName.contains("MySQL")
-                        || driverName.contains("MariaDB")
-                        || driverName.contains("H2")) {
-                    sql = SQLQueries.RETRIEVE_ACCESS_TOKEN_VALIDATION_DATA_IDP_NAME_MYSQL;
+            boolean isMySQLTypeDB = driverName.contains("MySQL") || driverName.contains("MariaDB")
+                    || driverName.contains("H2");
+            if (OAuth2ServiceComponentHolder.isAcrColumnEnabled()) {
```

Comment:
Why do we need `isAcrColumnEnabled` here?

In migration, this column should be created right

Reply:
Yes, but as we discussed, the binary should work with the old DB schema as well for backward compatibility. Once the migration is completed we can create another ticket and remove these if checks